File: Algorithms/Python3.x/1266-Minimum_Time_Visiting_All_Points.py
class Solution:
    def minTimeToVisitAllPoints(self, points: List[List[int]]) -> int:
        result = 0
        x1, y1 = points[0]
        for x2, y2 in points[1:]:
            result += max(abs(x1 - x2), abs(y1 - y2))
            x1, y1 = x2, y2
        return result

File: Algorithms/C++/1266-Minimum_Time_Visiting_All_Points.cpp
// Runtime: 12 ms, faster than 40.00% of C++ online submissions for Minimum Time Visiting All Points.
// Memory Usage: 10.1 MB, less than 100.00% of C++ online submissions for Minimum Time Visiting All Points.
class Solution {
public:
    int minTimeToVisitAllPoints(vector<vector<int>>& points) {
        int ans = 0;
        for (int i = 1; i < points.size(); ++i)
            ans += max(abs(points[i][0] - points[i - 1][0]), abs(points[i][1] - points[i - 1][1]));
        return ans;
    }
};
